src/Grpc.Net.Client/Balancer/Subchannel.cs

```diff
@@ -264,6 +266,11 @@ public void RequestConnection()
             }
         }
 
+        if (connectionRequested)
+        {
+            UpdateConnectivityState(ConnectivityState.Connecting, "Connection requested.");
```

Comment:
Is this now going to cause races where someone else comes along and modifies the Subchannel state and we end up updating it to `Connecting` when it should stay as `TransientFailure` or `Ready`?

Reply:
I believe this is fine. The logic in load balancing is designed that state can change at any point, and other parts of the system then react to it. All other places update the connectivity state outside the subchannel lock, this one is the outlier.